packages/providers/src/sdk/CryptKeeperInjectedProvider.ts

```diff
@@ -337,46 +336,49 @@ export class CryptKeeperInjectedProvider {
         signal,
         merkleStorageAddress,
         merkleProofArtifacts,
-        merkleProof,
+        merkleProofProvided,
       },
-      source: "offscreen",
-    }) as Promise<SemaphoreProof>;
+    }) as Promise<SemaphoreFullProof>;
   }
 
   /**
-   * Generates an RLN proof.
+   * Generates an RLN (Reputation Linked-Note) proof based on the provided RLN proof request.
    *
-   * @param {string} externalNullifier - The external nullifier.
-   * @param {string} signal - The signal.
-   * @param {string | MerkleProofArtifacts} merkleProofArtifactsOrStorageAddress - The merkle proof artifacts or storage address.
-   * @param {string} rlnIdentifier - The RLN identifier.
-   * @returns {Promise<RLNFullProof>} A Promise that resolves to the RLN proof.
+   * @param {string} rlnIdentifier - The RLN identifier as a string.
+   * @param {string} message - The message content as a string.
+   * @param {bigint | string} epoch - The epoch value as a bigint or string.
+   * @param {string | MerkleProofArtifacts} merkleProofArtifactsOrStorageAddress - Either the merkle proof artifacts or the merkle storage address as a string or MerkleProofArtifacts.
+   * @param {bigint | number} [messageLimit=1] - The message limit as a bigint or number. Defaults to 1 if not provided.
+   * @param {bigint | number} [messageId=0] - The ID of the message as a bigint or number. Defaults to 0 if not provided.
+   
+   * @returns {Promise<RLNSNARKProof>} A Promise that resolves to the generated RLN proof.
    */
-  async rlnProof(
-    externalNullifier: string,
-    signal: string,
-    merkleProofArtifactsOrStorageAddress: string | MerkleProofArtifacts,
-    rlnIdentifier: string,
-  ): Promise<RLNFullProof> {
+  async rlnProof({
+    rlnIdentifier,
+    message,
+    epoch,
+    merkleProofProvided,
+    merkleProofArtifactsOrStorageAddress,
+    messageLimit,
+    messageId,
+  }: Partial<IRlnProofRequest>): Promise<RLNSNARKProof> {
```

Comment:
I think all these params can't be optional. Right now it can be called like `rlnProof({})` and it would be a valid call during compilation time.